services/blockchain-app-registry/shared/metadata.js

```diff
@@ -181,16 +181,31 @@ const getBlockchainAppsMetadata = async (params) => {
 	}
 
 	if (params.isDefault !== true && blockchainAppsMetadata.data.length < params.limit) {
+		const originalOffset = params.offset;
+
+		// If params.isDefault is not passed as a parameter then adjust the offset
+		if (params.isDefault !== false) {
+			const totalDefaultApps = await applicationMetadataTable.count(
+				{ ...params, limit, isDefault: true },
+			);
+
+			params.offset = params.offset - totalDefaultApps > 0 ? params.offset - totalDefaultApps : 0;
+		}
+
 		const nonDefaultApps = await applicationMetadataTable.find(
 			{ ...params, limit, isDefault: false },
 			['network', 'appDirName', 'isDefault'],
 		);
 
+		// Reset the offset to the original value
+		params.offset = originalOffset;
+
 		blockchainAppsMetadata.data.push(...nonDefaultApps);
 	}
```

Comment:
I'd prefer to not modify the params.
```suggestion
	if (params.isDefault !== true && blockchainAppsMetadata.data.length < params.limit) {
		let offset = { params };

		// If params.isDefault is not passed in the request then adjust the offset
		if (!('isDefault' in params)) {
			const totalDefaultApps = await applicationMetadataTable.count(
				{ ...params, limit, isDefault: true },
			);

			offset = (params.offset - totalDefaultApps) > 0 ? params.offset - totalDefaultApps : 0;
		}

		const nonDefaultApps = await applicationMetadataTable.find(
			{ ...params, offset, limit, isDefault: false },
			['network', 'appDirName', 'isDefault'],
		);

		blockchainAppsMetadata.data.push(...nonDefaultApps);
	}
```